lib/tablex.js

```diff
@@ -1,13 +1,14 @@
 'use strict';
-module.exports = function (buffer){
-  var data = new DataView(buffer);
-  var offset = 8;
+module.exports = function (buffer, bufferx){
+  var tbl_num_rows = (new DataView(buffer)).getUint32(4,true);
+  var data = new DataView(bufferx);
   var rows = [];
-  var len = data.getUint32(offset, true);
-  offset += 8;//yes 8
+  var offset = 16;
   var i = 0;
-  while(i < len){
-    rows[i++] = data.getUint32(offset, true);
+  while(i<tbl_num_rows){
+    var val = data.getUint32(offset,true);
+    if(val==0){offset += 5;continue;}
```

Comment:
This would probably be better written as if (val ! == 0) { row[i++... 

Because the offset gets increased either way.